swift/looker/rtl/baseTransport.swift

```diff
@@ -99,14 +99,27 @@ class BaseTransport : ITransport  {
 
         var result: RequestResponse? = nil
         let task = self.session.dataTask(with: req!) { data, response, error in
-            let res = response as? HTTPURLResponse
             var sdkError: SDKError?
-            if let err = error, let r = res {
-                sdkError = SDKError(err.localizedDescription, code: r.statusCode)
-            }
-            if let r = res, !BaseTransport.ok(r) {
-                let dataString = String(data: data!, encoding: .utf8)
-                sdkError = SDKError(dataString ?? "Error with \(method) \(path)", code: r.statusCode)
+            if let err = error {
+                let msg = "\(method) \(path) error: \(err.localizedDescription)"
+                if let r = response as? HTTPURLResponse {
+                    sdkError = SDKError(msg, code: r.statusCode)
+                } else {
+                    if let d = data {
+                        sdkError = SDKError(msg, reason: String(data: d, encoding: .utf8))
+                    } else {
+                        sdkError = SDKError(msg, reason: "Unknown failure")
+                    }
+                }
+            } else {
+                if let r = response as? HTTPURLResponse {
+                    if !BaseTransport.ok(r) {
```

Comment:
You can combine these two to reduce nesting. But the execution result is slightly different...
```
if let r = response as? HTTPURLResponse, !BaseTransport.ok(r) {
```

Reply:
I'm just trying to eliminate all potential paths when compounded errors might result in an untrapped error condition. Without being able to reliably test that line I'm more comfortable leaving it broken out.